### PR TITLE
chore(EMS-3185): No PDF - Financial details - Separate encryption vectors into their own tables

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR updates the financial details encryption/decryption data model so that all vectors are in their own table.

Now, instead of having everything in the following 2x tables:

- `LossPayeeFinancialInternational`
- `LossPayeeFinancialUk`

We now have:
- `LossPayeeFinancialInternational`
- `LossPayeeFinancialInternationalVector`
- `LossPayeeFinancialUk`
- `LossPayeeFinancialUkVector`

This not only improves the data model and simplifies the existing table, but also helps add another security layer.

## Resolution :heavy_check_mark:
- Update SQL dump.
- Update `createALossPayeeFinancialInternational` helper function to create a vector relationship.
- Update `createALossPayeeFinancialUk` helper function to create a vector relationship.
- Update decryption helper functions to consume the vectors via a nested `vector` object.
- Simplify decryption mappings.
- Update `getPopulatedApplication` helper function to return vectors in the new structure.
- Update `mapLossPayeeFinancialDetailsInternational` and `mapLossPayeeFinancialDetailsUk` to return vectors in a nested object.
- Create new helper functions:
  - `updateLossPayeeFinancialInternationalVector`.
  - `updateLossPayeeFinancialUkVector`.
- Update Keystone schema.
- Update `createFullApplication` test helper function to generate vector relationships.
- Create new test helper functions:
  - `createLossPayeeFinancialDetailsInternationalVector`.
  - `createLossPayeeFinancialDetailsUkVector`
- Update test mocks.
- Update types.

## Miscellaneous :heavy_plus_sign:
- Replace all instace of "financial details" with just, "financial".
- Add logs and a try/catch handler to `decryptFinancialInternational` helper function, same as the UK helper.
- Update some `context` documentation.

### CODE REVIEW

File: src/api/helpers/decrypt-financial-international/index.test.ts [hunk 1]
```diff
@@ -36,97 +36,96 @@ describe('api/helpers/decrypt-financial-international', () => {
     });
   });
 
-  describe('when ibanVector is undefined', () => {
-    it('should return iban as an empty string', () => {
-      const variables = {
-        ...financialInternational,
-        ibanVector: undefined,
-      } as ApplicationLossPayeeFinancialInternational;
+  // describe('when ibanVector is undefined', () => {
```
Comment (ttbarnes) [inline src/api/helpers/decrypt-financial-international/index.test.ts:39]: TODO: update tests.

File: src/api/types/test-helpers/index.ts [hunk 2]
```diff
@@ -5,6 +5,12 @@ import { Account } from '../account';
 export interface TestHelperCreate {
   context: Context;
   companyId?: string;
+  // TODO
+  // TODO
+  // TODO
+  // TODO
+  // TODO: real type. for international/uk vector test helpers.
```
Comment (ttbarnes) [inline src/api/types/test-helpers/index.ts:12]: TODO: address this comment

File: src/api/custom-resolvers/mutations/update-loss-payee-financial-details-uk/index.test.ts [hunk 3]
```diff
@@ -36,11 +46,13 @@ describe('custom-resolvers/update-loss-payee-financial-details-uk', () => {
     });
   });
 
+  // TODO: same error handling as international.
```
Comment (ttbarnes) [inline src/api/custom-resolvers/mutations/update-loss-payee-financial-details-uk/index.test.ts:49]: TODO: improve unit test coverage

File: src/api/helpers/decrypt-nominated-loss-payee/index.test.ts [hunk 4]
```diff
@@ -20,30 +20,37 @@ describe('api/helpers/decrypt-nominated-loss-payee', () => {
...
 
       expect(decryptSpy).toHaveBeenCalledTimes(0);
     });
   });
 
-  describe('when "decryptFinancialUk" is "true"', () => {
+  // TODO:
+  // TODO:
+  // TODO:
+  // describe('when "decryptFinancialUk" is false', () => {
```
Comment (ttbarnes) [inline src/api/helpers/decrypt-nominated-loss-payee/index.test.ts:45]: TODO: improve unit test coverage.

File: src/api/helpers/update-loss-payee-financial-uk/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { Context, LossPayeeFinancialUkUpdateInput } from '.keystone/types'; // eslint-disable-line
+import updateLossPayeeFinancialUkUk from '.';
```
Comment (Zainzzkk) [inline src/api/helpers/update-loss-payee-financial-uk/index.test.ts:2]: typo?

File: src/api/helpers/update-loss-payee-financial-uk/index.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { Context, LossPayeeFinancialUkUpdateInput } from '.keystone/types'; // eslint-disable-line
+import { ApplicationLossPayeeFinancialUk } from '../../types';
+
+/**
+ * updateLossPayeeFinancialInternationalUk
+ * @param {Object} context: KeystoneJS context API
+ * @param {Object} id: Loss payee financial international uk ID
```
Comment (Zainzzkk) [inline src/api/helpers/update-loss-payee-financial-uk/index.ts:7]: String? and Context? above

File: src/api/package.json [hunk 7]
```diff
@@ -19,7 +19,7 @@
     "lint": "eslint ./",
     "lint:fix": "eslint ./ --fix",
     "start": "keystone start --no-ui",
-    "test": "jest --runInBand --verbose --silent --coverage --config=jest.config.js"
+    "test": "jest --runInBand --verbose --coverage --config=jest.config.js"
```
Comment (Zainzzkk) [inline src/api/package.json:22]: revert

**Review by Zainzzkk — COMMENTED** [review]
There are a few documentation places here where documentation for context has Object type, can these be changed to Context?

File: src/api/package.json (hunk 7 above)
Comment (ttbarnes) [reply src/api/package.json:22]: Whoops, great spot

**Comment by ttbarnes** [comment]
> There are a few documentation places here where documentation for context has Object type, can these be changed to Context?

Good shout, i've updated all instances of this in the API.

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**